src/jsep.js

```diff
@@ -305,48 +310,20 @@ export class Jsep {
 		return nodes;
 	}
 
-	//
 	/**
-	 * The main parsing function. Much of this code is dedicated to ternary expressions
+	 * The main parsing function.
 	 * @returns {?jsep.Expression}
 	 */
 	gobbleExpression() {
-		const test = this.gobbleBinaryExpression();
-
+		this.node = false;
```

Comment:
I know this pattern was commented on in #123 - it's basically trying to set an initial value, then it can check if it was set by the hook (without having a return value from multiple potential hooks). Any other ideas or suggestions?

Reply:
If you actually use an `env` object (with `context: this`) instead of `this`, you can just check for the presence of a given property. Not sure how generally useful `node` can be if read at random times and I can easily see bugs happening because some code forgot to reset it.